internal/pkg/agent/application/coordinator/config_patcher.go

```diff
@@ -0,0 +1,49 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package coordinator
+
+import (
+	"context"
+)
+
+type ConfigPatch func(change ConfigChange) ConfigChange
+
+// ConfigPatchManager is a decorator to restore some agent settings from the elastic agent configuration file
+type ConfigPatchManager struct {
+	inner  ConfigManager
+	outch  chan ConfigChange
+	patchf ConfigPatch
+}
+
+func (c ConfigPatchManager) Run(ctx context.Context) error {
+	go c.patch(c.inner.Watch(), c.outch)
+	return c.inner.Run(ctx)
+}
+
+func (c ConfigPatchManager) Errors() <-chan error {
+	return c.inner.Errors()
+}
+
+func (c ConfigPatchManager) ActionErrors() <-chan error {
+	return c.inner.ActionErrors()
+}
+
+func (c ConfigPatchManager) Watch() <-chan ConfigChange {
+	return c.outch
+}
+
+func (c ConfigPatchManager) patch(src <-chan ConfigChange, dst chan ConfigChange) {
+	for ccc := range src {
+		dst <- c.patchf(ccc)
+	}
+}
```

Comment:
This struct just applies patches to config returned by a ConfigManager.
Generating such a patch is task of the patchFn, so this can be used with any number of patching functions and the functions are easier to test

If the compiler inlines it, so be it :rofl: